Documentation/ceph-cluster-crd.md

```diff
@@ -1319,128 +1319,63 @@ The features available from the external cluster will vary depending on the vers
-
-**Example**:
-
-```console
-export NAMESPACE=rook-ceph-external
-export ROOK_EXTERNAL_FSID=3240b4aa-ddbc-42ee-98ba-4ea7b2a61514
-export ROOK_EXTERNAL_CEPH_MON_DATA=a=172.17.0.4:3300
-export ROOK_EXTERNAL_ADMIN_SECRET=AQC6Ylxdja+NDBAAB7qy9MEAr4VLLq4dCIvxtg==
-```
-
-If the Ceph admin key is not provided, the following script needs to be executed on a machine that can connect to the Ceph cluster using the Ceph admin key.
-On that machine, run:
-
-```sh
-. deploy/examples/create-external-cluster-resources.sh
-```
-
-The script will source all the necessary environment variables for you. It assumes the namespace name is `rook-ceph-external`.
-This can be changed by running the script like (assuming namespace name is `foo` this time):
-
-```sh
-ns=foo . deploy/examples/create-external-cluster-resources.sh
-```
-
-When done you can execute: `import-external-cluster.sh` to inject them in your Kubernetes cluster.
-
-> **WARNING**: Since only Ceph admin key can create CRs in the external cluster, please make sure that rgw pools have been prepared. You can get existing pools by running `ceph osd pool ls`.
+#### Run commands on a standalone [ceph cluster](https://docs.ceph.com/en/pacific/cephadm/#cephadm):
 
-**Example**:
-
-```console
-ceph osd pool ls
-```
->```
->my-store.rgw.control
->my-store.rgw.meta
->my-store.rgw.log
->my-store.rgw.buckets.index
->my-store.rgw.buckets.non-ec
->my-store.rgw.buckets.data
->```
+In order to configure an external Ceph cluster with Rook, we need to inject some information in order to connect to that cluster.
 
-In this example, you can simply export RGW_POOL_PREFIX before executing the script like this:
+1. Run the python script [create-external-cluster-resources.py](/deploy/examples/create-external-cluster-resources.py) for creating all users and keys.
 
 ```console
-export RGW_POOL_PREFIX=my-store
+python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --namespace <namespace> --format bash
 ```
 
-The script will automatically create users and keys with the lowest possible privileges and populate the necessary environment variables for `deploy/examples/import-external-cluster.sh` to work correctly.
-
-Finally, you can simply execute the script like this from a machine that has access to your Kubernetes cluster:
-
+rbd-data-pool-name should be created, and if you need to have CephFS secret details you have to pass filesystem name, and if you need to create rgw_admin_ops_user you need to pass rgw_endpoint (Rados GateWay endpoint (in <IP>:<PORT> format)) , and if you need to export rgw-pool-prefix prepare rgw pools
+example: 
 ```console
-bash deploy/examples/import-external-cluster.sh
+python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --cephfs-filesystem-name <filesystem-name> --rgw-endpoint  <rgw-endpoint> --namespace <namespace> --rgw-pool-prefix <rgw-pool-prefix> --format bash
```

Comment:
Let's replace the example command on line 1331 with this command that has all the options. Then we can have some bullets after it that explains each parameters, with required params first, then optional params. Something like this:

```console
python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --cephfs-filesystem-name <filesystem-name> --rgw-endpoint  <rgw-endpoint> --namespace <namespace> --rgw-pool-prefix <rgw-pool-prefix> --format bash
```

- `--namespace`: (Which namespace is this???)
- `--format bash`: The format of the output
- `--rbd-data-pool-name`:  (optional) The name of an rbd pool to create
- `--cephfs-filesystem-name`: (optional) The name of the filesystem 
- `--rgw-endpoint`: (optional) The RADOS Gateway endpoint in the format `<IP>:<PORT>`
- `--rgw-pool-prefix`: (optional) The prefix of the RGW pools, for example `default` (what is a common default for this???)